src/plugins/intel_npu/src/plugin/src/plugin.cpp

```diff
@@ -513,6 +517,64 @@ Plugin::Plugin()
             _supportedProperties.emplace_back(ov::PropertyName(property.first, std::get<1>(property.second)));
         }
     }
+
+    // Check if CID fallback is required
+    if (_backends->isCompilerFallbackRequired()) {
+        _logger.info("Driver Version is too old to use MLIR Compiler. "
+                     "Driver Compiler will be set as default compiler type.");
+        std::stringstream strStream;
+        strStream << ov::intel_npu::CompilerType::DRIVER;
+        _globalConfig.update({{std::string(ov::intel_npu::compiler_type.name()), strStream.str()}});
+    } else {
+        // Compare CIP and Driver versions for ELF and MI
+        const auto cip = createCompiler(ov::intel_npu::CompilerType::MLIR, _logger);
+        // The driver versions are fetched from zero device
+        const auto device = _backends->getDevice();
+        if (device != nullptr) {
```

Comment:
Done, although this will bring a possible future issue, I've added a comment explaining the problem.

`// Instructions added below this point could be skipped due to early return`

Reply:
Moved the code in a separate method so we can return without issues.